File: fmarket/src/main/java/ro/fmarket/admin/account/company/CompanySearchObject.java
package ro.fmarket.admin.account.company;

import java.util.List;

import javax.validation.constraints.NotNull;

import org.hibernate.validator.constraints.Email;

import lombok.Data;

@Data
public class CompanySearchObject {

	@NotNull
	private Integer page;
	
	private Integer accountId;
	
	private Integer companyId;
	
	private String name;
	
	@Email
	private String email;
	
	private Integer companyDomain; //TODO should be ID
	
	private List<Integer> demandDomains; //TODO also IDS
	
}

File: fmarket/src/main/webapp/app/models/companySearchObject.ts
import { Select2Item } from './../components/selectComponent/selectComponent';
/**
 * Created by NicolaeB on 5/26/2016.
 */
export class CompanySearchObject{
    name:string = "";
    companyDomain:Select2Item;
    demandDomains:Array<Select2Item> = [];
    page:number =1;
}